### PR TITLE
Remove duplication from `UserFacingPrinter`

### PR DESCRIPTION
**Pull request by felixmulder**
@smarter: how far do we want to take this? Should I try to get rid of `UserFacingPrinter` completely?

I think there might be some things that won't print as expected if we do that. But I could try, anyway - up to you.

### CODE REVIEW

**Comment by smarter** [comment]
I probably won't have time to review this week, you may want to ask someone else.

> how far do we want to take this? Should I try to get rid of UserFacingPrinter completely?

The less stuff we have the better it is I think, if a separate printer is needed for the REPL it needs to be well justified.

Also in https://github.com/lampepfl/dotty/pull/3032/commits/e331f87642a10e80db4cd931476bfd0ada92d9f3 you put `@smarter` in the commit message which means that everytime you re-push this commit I get an email, please avoid that :).

File: project/Build.scala [hunk 1]
```diff
@@ -358,47 +359,6 @@ object Build {
...
-          case (beforeCp, Nil) => beforeCp ++ ("-classpath" :: classPath :: Nil)
-          case (beforeCp, rest) => beforeCp ++ rest
-        }
-
-        (runMain in Compile).toTask(
-          s" dotty.tools.repl.Main " + fullArgs.mkString(" ")
-        )
-      }.evaluated,
-
-      repl := run.evaluated,
```
Comment (smarter) [inline project/Build.scala]: Missing a definition `repl :=` in dotty-compiler now.

Comment (felixmulder) [reply project/Build.scala]: Fixed in #3026 :)

File: compiler/src/dotty/tools/dotc/printing/UserFacingPrinter.scala [hunk 2]
```diff
@@ -0,0 +1,71 @@
+package dotty.tools.dotc
+package printing
+
+import core._
+import Constants.Constant, Contexts.Context, Denotations._, Flags._, Names._
+import NameOps._, StdNames._, Decorators._, Scopes.Scope, Types._, Texts._
+import SymDenotations.NoDenotation, Symbols.{ Symbol, ClassSymbol, defn }
+
+class UserFacingPrinter(_ctx: Context) extends RefinedPrinter(_ctx) {
```
Comment (OlivierBlanvillain) [inline compiler/src/dotty/tools/dotc/printing/UserFacingPrinter.scala]: Do you use `_ctx` because `ctx` is already defined in `RefinedPrinter`?

**Comment by OlivierBlanvillain** [comment]
It's great to have these things merged together, LGTM!

File: build.sbt [hunk 3]
```diff
@@ -18,7 +18,6 @@ val `dotty-bench-bootstrapped` = Build.`dotty-bench-bootstrapped`
 val `scala-library` = Build.`scala-library`
 val `scala-compiler` = Build.`scala-compiler`
 val `scala-reflect` = Build.`scala-reflect`
-val `dotty-repl` = Build.`dotty-repl`
```
Comment (OlivierBlanvillain) [inline build.sbt]: Is it intentional to keep the `dotty-repl` project in Build.scala?

File: compiler/src/dotty/tools/dotc/printing/UserFacingPrinter.scala (hunk 2 above)
Comment (felixmulder) [reply compiler/src/dotty/tools/dotc/printing/UserFacingPrinter.scala]: Yes and that's what we do in RefinedPrinter when extending plain printer :)

File: build.sbt (hunk 3 above)
Comment (felixmulder) [reply build.sbt]: Looks like an oversight when rebasing.

Comment (OlivierBlanvillain) [reply build.sbt]: (Remove it in #3044)

**Pull request merged**